packages/backend-for-frontend/src/services/toolService.ts

```diff
@@ -0,0 +1,465 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  ApiKey,
+  ClientAssertion,
+  ConsumerKey,
+  FailedValidation,
+  SuccessfulValidation,
+  validateClientKindAndPlatformState,
+  validateRequestParameters,
+  verifyClientAssertion,
+  verifyClientAssertionSignature,
+} from "pagopa-interop-client-assertion-validation";
+import {
+  AgreementId,
+  ApiError,
+  ClientId,
+  EServiceId,
+  ItemState,
+  PurposeId,
+  TenantId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { getAllFromPaginated, WithLogger } from "pagopa-interop-commons";
+import {
+  agreementApi,
+  authorizationApi,
+  bffApi,
+  catalogApi,
+  purposeApi,
+} from "pagopa-interop-api-clients";
+import { BffAppContext } from "../utilities/context.js";
+import {
+  activeAgreementByEserviceAndConsumerNotFound,
+  agreementDescriptorNotFound,
+  agreementNotFound,
+  clientAssertionPublicKeyNotFound,
+  ErrorCodes,
+  eserviceDescriptorNotFound,
+  missingActivePurposeVersion,
+  multipleAgreementForEserviceAndConsumer,
+  organizationNotAllowed,
+  purposeIdNotFoundInClientAssertion,
+} from "../model/errors.js";
+import { TokenGenerationValidationStepFailure } from "../../../api-clients/dist/bffApi.js";
+import {
+  PagoPAInteropBeClients,
+  AgreementProcessClient,
+  CatalogProcessClient,
+} from "../clients/clientsProvider.js";
+import { getLatestAgreement } from "./agreementService.js";
+
+export function toolsServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    async validateTokenGeneration(
+      clientId: string | undefined,
+      clientAssertion: string,
+      _clientAssertionType: string,
+      _grantType: string,
+      ctx: WithLogger<BffAppContext>
+    ): Promise<bffApi.TokenGenerationValidationResult> {
+      ctx.logger.info(`Validating token generation for client ${clientId}`);
+
+      const { errors: parametersErrors } = validateRequestParameters({
+        client_assertion: clientAssertion,
+        client_assertion_type:
+          "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+        grant_type: "client_credentials",
```

Comment:
I think this still needs to be addressed from the previous discussion. Even if these are the only values allowed in the client assertion logic, let's pass the two parameters from the FE. In this way, the user will get an error in case of unexpected values, and we don't have to change this code in case new accepted values are added in the future.